porch/pkg/registry/porch/packagecommon.go

```diff
@@ -141,30 +145,71 @@ func (r *packageCommon) updatePackageRevision(ctx context.Context, name string,
 
 	oldPackage, err := r.getPackage(ctx, name)
 	if err != nil {
-		return nil, false, err
+		if apierrors.IsNotFound(err) {
+			// For server-side apply, we create new objects where they don't already exist
```

Comment:
should we be checking `forceAllowCreate` here to see if creation on update is disabled?

(I honestly don't know whether the `false` value means "do not create on update" or  whether it means "do not force" which seems to read very permissive "just do whatever :)")

Reply:
Good idea - I added a check for it, and it seems to work as expected.

I agree it's not very clear what the intent of the flag is...